File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        n=len(matrix)
        m=len(matrix[0])
        low=0;
        high=(n*m-1)
        while(low<=high):
            mid=(low+high)/2
            row=mid//m
            col=mid%m
            if(matrix[row][col]==target):
                return True
            if(matrix[row][col]<target):
                low=mid+1
            if(matrix[row][col]>target):
                high=mid-1
        return False
        

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n=matrix.size();
        int m=matrix[0].size();
        int low=0;
        int high=(n*m -1);
        while(low<=high)
        {
            int mid=(low+high)/2;
            int row=mid/m;
            int col=mid%m;
            if(target==matrix[row][col])
            {
                return true;
            }
            if(target<matrix[row][col])
            {
                high=mid-1;
            }
            if(target>matrix[row][col])
            {
                low=mid+1;
            }
        }
        return false;
    }
};